### PR TITLE
Refactor

### PR DESCRIPTION
**Pull request by yuehhua**

### CODE REVIEW

File: src/gdrive.jl [hunk 1]
```diff
@@ -19,7 +23,7 @@ function unshortlink(url; kw...)
     return url
 end
 
-function gsheet_handler(url; format=:csv)
+function prepare_gsheet_url(url; format=:csv)
```
Comment (foldfelis) [inline src/gdrive.jl:26]: This is not a preparation, semantic mismatch.

File: src/gdrive.jl [hunk 2]
```diff
@@ -30,26 +34,28 @@ function gsheet_handler(url; format=:csv)
     return url
 end
 
-function gfile_handler(url)
+function prepare_gfile_url(url)
```
Comment (foldfelis) [inline src/gdrive.jl:37]: This is not a preparation, semantic mismatch.

File: src/gdrive.jl [hunk 3]
```diff
@@ -30,26 +34,28 @@ function gsheet_handler(url; format=:csv)
...
     # https://drive.google.com/file/d/<hash>/view?usp=sharing
     h = match(r"/file/d/([^\/]+)/", url)
-    (h === nothing) && throw("Can't find goole drive file ID in the url")
-
-    return "https://docs.google.com/uc?export=download&id=$(h.captures[])"
+    if isnothing(h)
+        throw(ErrorException("Can't find goole drive file ID in the url"))
+    else
+        return "https://docs.google.com/uc?export=download&id=$(h.captures[])"
+    end
```
Comment (foldfelis) [inline src/gdrive.jl:45]: How about 

```julia
isnothing(h) && throw(ErrorException("Can't find google drive file ID in the url"))

return "https://docs.google.com/uc?export=download&id=$(h.captures[])"
```

So as to match the style

File: src/gdrive.jl [hunk 4]
```diff
@@ -30,26 +34,28 @@ function gsheet_handler(url; format=:csv)
...
 end
 
-function find_gcode(cookies)
+function fetch_gcode(cookies)
```
Comment (foldfelis) [inline src/gdrive.jl:48]: I'm not sure if this is a `fetch`. To me this is more like a `find`.

File: src/gdrive.jl [hunk 5]
```diff
@@ -30,26 +34,28 @@ function gsheet_handler(url; format=:csv)
...
     for cookie in cookies
         (match(r"download_warning_", cookie.name) !== nothing) && (return cookie.value)
     end
 
     return
 end
 
-function find_filename(header)
+function fetch_filename(header)
```
Comment (foldfelis) [inline src/gdrive.jl:56]: I'm not sure if this is a `fetch`. To me this is more like a `find`.

File: src/gdrive.jl [hunk 6]
```diff
@@ -62,9 +68,9 @@ end
 function download_gdrive(url, localdir)
     cookiejars = Dict{String, Set{HTTP.Cookies.Cookie}}()
     HTTP.request("GET", url; cookies=true, cookiejar=cookiejars)
-    gcode = find_gcode(cookiejars["docs.google.com"])
+    gcode = fetch_gcode(cookiejars["docs.google.com"])
```
Comment (foldfelis) [inline src/gdrive.jl:71]: I'm not sure if this is a `fetch`. To me this is more like a `find`.

File: src/gdrive.jl [hunk 7]
```diff
@@ -77,10 +83,10 @@ function download_gdrive(url, localdir)
         header = HTTP.header(response, "Content-Disposition")
         isempty(header) && return
 
-        filepath = joinpath(localdir, find_filename(header))
+        filepath = joinpath(localdir, fetch_filename(header))
```
Comment (foldfelis) [inline src/gdrive.jl:86]: I'm not sure if this is a `fetch`. To me this is more like a `find`.

File: src/gdrive.jl [hunk 8]
```diff
@@ -108,11 +114,11 @@ function gdownload(url, localdir)
     url = unshortlink(url)
 
     if is_gfile(url)
-        url = gfile_handler(url)
+        url = prepare_gfile_url(url)
     elseif is_gsheet(url)
-        url = gsheet_handler(url)
+        url = prepare_gsheet_url(url)
```
Comment (foldfelis) [inline src/gdrive.jl:119]: This is not a preparation, semantic mismatch.

File: src/gdrive.jl (hunk 3 above)
Comment (yuehhua) [reply src/gdrive.jl:45]: To clarify the semantics, `if...else` clause expresses better.

Comment (foldfelis) [reply src/gdrive.jl:45]: The way I mentioned match the `Return Early Pattern` that make the return more explicitly for the function.

**Comment by chengchingwen** [comment]
Sorry, but we are fine with the current naming and coding style.

**Pull request closed without merging**